src/libcsg/topology.cc

```diff
@@ -261,25 +256,12 @@ void Topology::CheckMoleculeNaming(void) {
   }
 }
 
-void Topology::AddBondedInteraction(Interaction *ic) {
-  map<string, Index>::iterator iter;
-  iter = _interaction_groups.find(ic->getGroup());
-  if (iter != _interaction_groups.end()) {
-    ic->setGroupId((*iter).second);
-  } else {
-    Index i = _interaction_groups.size();
-    _interaction_groups[ic->getGroup()] = i;
-    ic->setGroupId(i);
-  }
-  _interactions.push_back(ic);
-  _interactions_by_group[ic->getGroup()].push_back(ic);
-}
-
-std::list<Interaction *> Topology::InteractionsInGroup(const string &group) {
-  map<string, list<Interaction *>>::iterator iter =
+std::list<const Interaction *> Topology::InteractionsInGroup(
+    const string &group) const {
+  map<string, list<const Interaction *>>::const_iterator iter =
       _interactions_by_group.find(group);
   if (iter == _interactions_by_group.end()) {
-    return list<Interaction *>();
+    return list<const Interaction *>();
```

Comment:
I would prefer std::vector simply for performance reasons but this PR is valuable in its current form